server/src/main/java/hu/flowacademy/epsilon/sport_event_organizer/service/CupService.java

```diff
@@ -0,0 +1,126 @@
+package hu.flowacademy.epsilon.sport_event_organizer.service;
+
+import hu.flowacademy.epsilon.sport_event_organizer.exception.CupNotFoundException;
+import hu.flowacademy.epsilon.sport_event_organizer.model.Cup;
+import hu.flowacademy.epsilon.sport_event_organizer.model.Team;
+import hu.flowacademy.epsilon.sport_event_organizer.model.User;
+import hu.flowacademy.epsilon.sport_event_organizer.repository.CupRepository;
+import hu.flowacademy.epsilon.sport_event_organizer.repository.TeamRepository;
+import hu.flowacademy.epsilon.sport_event_organizer.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Set;
+
+@Service
+@Transactional
+public class CupService {
+    @Autowired
+    private CupRepository cupRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private TeamRepository teamRepository;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private TeamService teamService;
+
+    public Cup save(Cup cup) {
+        User currentUser = userService.getCurrentUser().orElse(null);
```

Comment:
Ne legyen orElse(null), dobjon inkább UserNotFound kivételt

Reply:
Itt a getCurrentUser-t is lehet módosítani, hogy a visszatérése User legyen, ne optional és a függvényen belül dobjon hibát, ha nincs user

Reply:
Viszont akkor ne UserNotFound legyen, hanem valami Forbiden Exception